pkg/api/server/server.go

```diff
@@ -254,10 +242,42 @@ func (s *APIServer) Serve() error {
 	return <-errChan
 }
 
+// setupPprof enables pprof default endpoints
+// Note: These endpoints and the podman flag --cpu-profile are mutually exclusive
+//
+// Examples:
+// #1 go tool pprof -http localhost:8889 localhost:8888/debug/pprof/heap?seconds=120
+// Note: web page will only render after a sample has been recorded
+// #2 curl http://localhost:8888/debug/pprof/heap > heap.pprof && go tool pprof heap.pprof
+// FIXME: Make port and interface configurable
+func (s *APIServer) setupPprof() {
+	// Only enable profiling endpoints when debugging
+	if !logrus.IsLevelEnabled(logrus.DebugLevel) {
+		return
+	}
+
+	logrus.Infof("pprof endpoint is available on localhost:8888")
+	go func() {
+		old := runtime.SetMutexProfileFraction(1)
+		defer runtime.SetMutexProfileFraction(old)
+
+		runtime.SetBlockProfileRate(1)
+		defer runtime.SetBlockProfileRate(0)
+
+		router := mux.NewRouter()
+		router.PathPrefix("/debug/pprof/").HandlerFunc(pprof.Index)
+
+		err := http.ListenAndServe("localhost:8888", router)
```

Comment:
Why create a separate socket? I think we should reuse the server one otherwise we block port 8888 for no reason.